src/include/planner/operator/logical_aggregate.h

```diff
@@ -27,14 +27,16 @@ class LogicalAggregate final : public LogicalOperator {
     static constexpr LogicalOperatorType operatorType_ = LogicalOperatorType::AGGREGATE;
 
 public:
+    // TODO(Royi) make sure cardinality is being set correctly
     LogicalAggregate(binder::expression_vector keys, binder::expression_vector aggregates,
         std::shared_ptr<LogicalOperator> child)
-        : LogicalOperator{operatorType_, std::move(child)}, keys{std::move(keys)},
-          aggregates{std::move(aggregates)} {}
+        : LogicalOperator{operatorType_, std::move(child), aggregates.size()},
```

Comment:
I'm assuming here that the cardinality of the aggregate is `aggregates.size()`, is this correct?

Reply:
updated, cardinality will be 1 if `keys.size() == 0` and we'll take the child's card otherwise. Left a TODO to improve this estimation.